packages/amplify-category-api/src/graphql-transformer/transform-graphql-schema-v2.ts

```diff
@@ -191,6 +189,21 @@ const buildAPIProject = async (
   }
 
   const builtProject = await _buildProject(opts);
+  const apiKeyStatus = await getApiKeyStatus(context);
+  const apiKeyEntry: AppSyncAuthConfigurationAPIKeyEntry = [
+    opts.authConfig.defaultAuthentication, ...opts.authConfig.additionalAuthenticationProviders,
+  ].find(
+    (mode: AppSyncAuthConfigurationEntry) => mode.authenticationType === AuthorizationType.API_KEY && mode.apiKeyConfig,
+  ) as AppSyncAuthConfigurationAPIKeyEntry;
+  const config = apiKeyEntry?.apiKeyConfig;
+  const apiKeyId = `${(config as any)?.name || 'Default'}ApiKey`;
```

Comment:
Not necessarily, in most cases the API key config does not define the name, I didn't even see it on the type. It won't compile without the cast because the property doesn't exist, but other uses of API key config are referencing the name property because they aren't type safe, so for proper feature parity I still use it